module/dkg/broker.go

```diff
@@ -104,17 +107,22 @@ func (b *Broker) Broadcast(data []byte) {
 	if err != nil {
 		b.log.Fatal().Err(err).Msg("failed to create broadcast message")
 	}
-	success := b.retry(
-		func() error {
-			return b.dkgContractClient.Broadcast(bcastMsg)
-		},
-		RETRY_MAX,
-		RETRY_MILLISECONDS,
-	)
-	if !success {
-		b.log.Error().Msg("failed to broadcast message")
+
+	expRetry, err := retry.NewExponential(RETRY_MILLISECONDS)
+	if err != nil {
+		b.log.Fatal().Err(err).Msg("create retry mechanism")
+	}
+	maxedExpRetry := retry.WithMaxRetries(RETRY_MAX, expRetry)
+
+	err = retry.Do(context.TODO(), maxedExpRetry, func(ctx context.Context) error {
```

Comment:
Given the API, I feel like the most reasonable thing to do here is to use `context.Background()` (likewise for the other instances of `context.TODO()`)